chassis/core/src/main/java/com/griddynamics/jagger/rest/MasterRestController.java

```diff
@@ -32,168 +37,124 @@
-            TestGroupInfo testGroupInfo = new TestGroupInfo();
-            testGroupInfo.leadingTests = Lists.newArrayList();
-            for (Task leadingTask : compositeTask.getLeading()) {
-                testGroupInfo.leadingTests.add(getTestInfoFrom(leadingTask));
-            }
-    
-            testGroupInfo.attendantTests = Lists.newArrayList();
-            for (Task attendantTask : compositeTask.getAttendant()) {
-                testGroupInfo.attendantTests.add(getTestInfoFrom(attendantTask));
-            }
-    
-            testInfo = testGroupInfo;
-        }
-    
-        testInfo.name = task.getTaskName();
-        testInfo.sessionId = sessionIdProvider.getSessionId();
-        if (task.getNumber() > 0) {  // 0 means it wasn't set yet.
-            testInfo.number = task.getNumber();
-            testInfo.id = taskIdProvider.stringify(task.getNumber());
-            testInfo.status = taskExecutionStatusProvider.getStatus(testInfo.id);
-        }
+    private TestEntity convertFrom(WorkloadTask task) {
+        TestEntity testEntity = new TestEntity();
+        testEntity.setName(task.getName());
+        testEntity.setDescription(task.getDescription());
+        testEntity.setTestGroupIndex(task.getGroupNumber());
         
-        return testInfo;
-    }
-    
-    public static class TestGroupInfo extends TestInfo {
-        private List<TestInfo> leadingTests;
-        private List<TestInfo> attendantTests;
+        testEntity.setStartDate(convertFrom(task.getStartDate()));
+        testEntity.setEndDate(convertFrom(task.getEndDate()));
         
-        public List<TestInfo> getLeadingTests() {
-            return leadingTests;
-        }
+        testEntity.setLoad(task.getClock().toString());
+        testEntity.setClockValue(task.getClock().getValue());
+        testEntity.setTerminationStrategy(task.getTerminateStrategyConfiguration().toString());
+        testEntity.setTestExecutionStatus(Decision.FATAL);
         
-        public List<TestInfo> getAttendantTests() {
-            return attendantTests;
-        }
+        testEntity.setTestExecutionStatus(convertFrom(task.getStatus()));
+        
+        return testEntity;
     }
     
-    public static class TestInfo {
-        private String id;
-        private String sessionId;
-        private String name;
-        private Integer number;
-        private TaskData.ExecutionStatus status;
-        
-        public String getId() {
-            return id;
-        }
-        
-        public String getSessionId() {
-            return sessionId;
-        }
-        
-        public String getName() {
-            return name;
-        }
+    private Date convertFrom(Long millis) {
+        return millis != null ? new Date(millis) : null;
+    }
+    
+    private Decision convertFrom(TaskData.ExecutionStatus status) {
         
-        public Integer getNumber() {
-            return number;
+        if (status == null) {
```

Comment:
It's better to have only 1 return statement in method, otherwise code becomes more complicated to read and debug.